src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -390,13 +416,53 @@ private static MappedTranscriptIdInfo createMappedTranscriptIdInfo( final SAMSeq
             }
         }
 
+        // TODO: This may not be correct.  It's not clear that the whole sequence should be taken.  Perhaps it should be handled with logic downstream...
+        // If no CDS was specified, we use the whole sequence:
+        if ( transcriptIdInfo.codingSequenceStart == 0 ) {
+            transcriptIdInfo.codingSequenceStart = 1;
+        }
+        if ( transcriptIdInfo.codingSequenceEnd == 0 ) {
+            transcriptIdInfo.codingSequenceEnd = sequence.getSequenceLength();
+        }
+
         transcriptIdInfo.mapKey = sequence.getSequenceName();
         transcriptIdInfo.has3pUtr = has3pUtr;
         transcriptIdInfo.has5pUtr = has5pUtr;
 
         return transcriptIdInfo;
     }
 
+    /**
+     * Get the whole sequence from the GENCODE Transcript FASTA file for a given {@code transcriptId}.
+     * This will get ALL bases in the sequence of the given {@code transcriptId}, including any UTRs and the coding region.
+     * @param transcriptId The ID of the transcript to get from the FASTA file.
+     * @param transcriptIdMap A map from transcriptId to MappedTranscriptIdInfo, which tells us how to pull information for the given {@code transcriptId} out of the given {@code transcriptFastaReferenceDataSource}.
+     * @param transcriptFastaReferenceDataSource A {@link ReferenceDataSource} for the GENCODE transcript FASTA file.
+     * @return The whole sequence for the given {@code transcriptId} as represented in the GENCODE transcript FASTA file.
+     */
+    private static String getWholeSequenceFromTranscriptFasta( final String transcriptId,
```

Comment:
this method seems to be unused

Reply:
Yeah - it's a left-over idea I had.  I want to be able to get the sequences from the list of exons and the reference itself, rather than requiring the gencode sequence fasta file.

That said, it never worked properly so I'll take it out.